db/index.js

```diff
@@ -0,0 +1,73 @@
+const pg = require('pg')
+let confFile = './db_settings.json'
+const setting = require(confFile)
+const validate = require('./validate')
+
+const dbconfig = setting.dbconfig
+
+let conString = {
+  user: dbconfig.username,
+  database: dbconfig.dbname,
+  password: dbconfig.password,
+  host: dbconfig.dbhost,
+  port: dbconfig.dbport,
+  max: 10,
+  idleTimeoutMillis: 30000
+}
+
+let pool = new pg.Pool(conString)
+
+pool.connect(function (err, client, done) {
+  if (err) {
+    if (err.code === '3D000') {
+      console.log('Database does not exist, creating custom db')
+      const db = require('./db_initialise')
+      db.initialize(function (err) {
+        console.log(err)
+      })
+    } else if (err) {
+      console.log(err)
+    }
+  }
+  done()
+})
+
+exports.createCourse = (course, cb) => {
+  pool.connect(function (err, client, done) {
+    if (err) {
+      console.log(err)
+      cb(null)
+    }
+    if (validate.courseDetails(course)) {
+        client.query("INSERT INTO courses (coursename, startdate, enddate) VALUES ('"+course.coursename+"', '"+course.startdate+"', '"+course.enddate+"')", function (err) {
```

Comment:
Move all SQL into a data access layer. Preferably, look for a tool which allows the creation of ad-hoc queries without string concatenation. If you concatenate strings into SQL, you are always (always) susceptible to SQL injection attacks.